src/google/protobuf/repeated_field.h

```diff
@@ -346,7 +346,7 @@ class RepeatedField final {
   int total_size_;
   // Pad the Rep after arena allow for power-of-two byte sizes when
   // sizeof(Element) > sizeof(Arena*). eg for 16-byte objects.
-  static constexpr size_t kRepHeaderSize =
+  static const size_t kRepHeaderSize =
```

Comment:
Actually could you try changing this to `PROTOBUF_CONSTEXPR const`? I believe that should make this constexpr only for recent enough compilers. (And then we want `const` added on there so that it's always const no matter what.)